spec/models/service_template_spec.rb

```diff
@@ -285,6 +287,28 @@
         expect(ServiceTemplate.find_by(:name => "drew").guid).not_to eq(@st1.guid)
       end
 
+      it "with custom button" do
+        custom_button
+        expect(@st1.custom_buttons.count).to eq(1)
+        @st1.template_copy("drew")
+        expect(ServiceTemplate.count).to eq(2)
+        expect(ServiceTemplate.find_by(:name => "drew")).not_to be(nil)
+        expect(ServiceTemplate.find_by(:name => "drew").display).to be(false)
+        expect(ServiceTemplate.find_by(:name => "drew").guid).not_to eq(@st1.guid)
+        expect(ServiceTemplate.find_by(:name => "drew").custom_buttons.count).to eq(2)
+      end
+
+      it "with custom button set" do
+        custom_button_set.add_member(custom_button)
+        expect(@st1.custom_button_sets.count).to eq(1)
+        @st1.template_copy("drew")
+        expect(ServiceTemplate.count).to eq(2)
+        expect(ServiceTemplate.find_by(:name => "drew")).not_to be(nil)
```

Comment:
For better performance, can you call `find_by` once and reuse the result in the expectations?

```ruby
new_template = ServiceTemplate.find_by(:name => "drew")
expect(new_template.display)...
expect(new_template.guid)...
...
```

Reply:
Also, the `not_to be(nil)` feels unnecessary if the following expectations will test attributes of the record that was looked up.